virtualenv.py

```diff
@@ -100,12 +100,35 @@ def get_installed_pythons():
         import _winreg as winreg
 
     def get_installed_pythons():
+        final_exes = dict()
+
+        # Grab exes from 32-bit registry view
+        exes = _get_installed_pythons_for_view("-32", winreg.KEY_WOW64_32KEY)
+        # For 64-bit OS grab exes from 64-bit registry view
+        if sys.maxsize > 2**32 or "PROGRAMFILES(X86)" in os.environ:
```

Comment:
This check is a bit hackish:
* `sys.maxsize > 2**32` checks if Python running is 64-bit. This can be replaced with [`platform.architecture`](1)`()[0] == "64bit"`, which one is prefered?
* `"PROGRAMFILES(X86)" in os.environ` will fail if someone creates that environment variable on 32-bit Windows (you get each version twice) or deletes it on 64-bit one (you get 32-bit versions only).

[1]: https://docs.python.org/3/library/platform.html#platform.architecture

Reply:
Do we need to check? Can't we just try to collect, and ignore if it fails?

Reply:
The thing is it won't fail. On 32-bit system when trying to open 64-bit registry view it will still open 32-bit view. Versions will be falsely labelled as 64-bit or worse, we will get something like `3.5-32-64`.

I think comparing values of dicts returned from `_get_installed_pythons_for_view` would work. If they are the same, skip the second dict. Would that be better?

Reply:
I would then go for architecture via platform 👌

Reply:
`platform.architecture()[0]` will be `32bit` when running 32-bit Python on 64-bit OS, so `"PROGRAMFILES(X86)" in os.environ` is still needed.

How about this:
```python
exes_64 = _get_installed_pythons_for_view("-64", winreg.KEY_WOW64_64KEY)
if set(exes.values()) != set(exes_64.values()):
    exes.update(exes_64)
```

Reply:
feels good